gapir/client/session.go

```diff
@@ -62,14 +63,16 @@ func newSession(d bind.Device) *session {
 
 func (s *session) init(ctx context.Context, d bind.Device, abi *device.ABI, launchArgs []string) error {
 	defer close(s.inited)
-
 	var err error
+
 	if host.Instance(ctx).SameAs(d.Instance()) {
-		err = s.newHost(ctx, d, launchArgs)
-	} else if d, ok := d.(adb.Device); ok {
-		err = s.newADB(ctx, d, abi)
+		err = s.newHost(ctx, d, abi, launchArgs)
```

Comment:
Again, it's a shame with the device abstractions that host and remote can't share the same code.

Reply:
Agreed.
I think I might be able to clean this up a bit. I wanted to see how the gapit changes shook out before I tried unifying things like this.  Since they will be similar in nature.